codescantask/policies/undeclared-policy-check.ts

```diff
@@ -42,66 +38,32 @@ export class UndeclaredPolicyCheck extends PolicyCheck {
         super(`Undeclared Policy`);
     }
 
-    async run(scannerResults: ScannerResults): Promise<void> {
-        await this.start();
-
-        const nonDeclaredComponents: Component[] = [];
-        let declaredComponents: Partial<Component>[] = [];
-
-        const comps = getComponents(scannerResults);
-
-
-        try {
-          const sbom = await parseSBOM(SBOM_FILEPATH);
-          declaredComponents = sbom.components || [];
-        } catch (e: unknown) {
-            if (e instanceof Error) {
-                tl.warning(`Warning on policy check: ${this.checkName}. SBOM file cannot be parsed or was not found.`);
-            }
-        }
-
-        comps.forEach(c => {
-            if (!declaredComponents.some(component => component.purl === c.purl)) {
-                nonDeclaredComponents.push(c);
-            }
-        });
-
-        const summary = this.getSummary(nonDeclaredComponents);
-        const details = this.getDetails(nonDeclaredComponents);
 
-        if (details) {
-            await this.uploadArtifact(this.policyCheckResultName, details);
-        }
-
-        if (nonDeclaredComponents.length === 0) {
-            await this.success(summary, details);
-        } else {
-            await this.reject(summary, details);
-        }
+    private buildArgs(): Array<string> {
+        return ['run', '-v', `${REPO_DIR}:/scanoss`, RUNTIME_CONTAINER, 'inspect', 'undeclared', '--input',
+            OUTPUT_FILEPATH, '--format', 'md',
+            ...(!SCANOSS_SETTINGS ? ['--sbom-format', 'legacy']: []) // Sets sbom format output to legacy if SCANOSS_SETTINGS is not true
```

Comment:
What happens if default `SBOM_FILEPATH` does not exists and `SCANOSS_SETTINGS` is false? We should pass `--sbom-format` anyways?

Reply:
If the SBOM_FILEPATH has a default value set. If the file is not found and the SCANOSS_SETTINGS is set to false, we use the legacy sbom format